api/v1beta1/oscmachine_validation.go

```diff
@@ -134,9 +148,9 @@ func ValidateSubregionName(subregionName string) (string, error) {
 // ValidateDeviceName check that DeviceName  is a valid DeviceName
 func ValidateDeviceName(deviceName string) (string, error) {
 	last := deviceName[len(deviceName)-1:]
-	isValidateDeviceName := regexp.MustCompile(`^[a-z]$`).MatchString
+	isValidateDeviceName := regexp.MustCompile(`^[0-9a-z]$`).MatchString
 	switch {
-	case strings.HasPrefix(deviceName, "/dev/xvd") && len(deviceName) == 9 && isValidateDeviceName(last):
+	case (strings.Contains(deviceName, "/dev/xvd") || strings.Contains(deviceName, "/dev/sda")) && len(deviceName) == 9 && isValidateDeviceName(last):
```

Comment:
if / else would be ok here.

You can combine all the logic in one regex in one human-readable regex I think

Also, you theoretically have "xvdaa" once "xvdz" is taken

Reply:
Ok done